### ISSUE TITLE
Package.exports types file specification missing

### ISSUE THREAD

**Issue by mattstyles** [svg/svgo#1993]
**Describe the bug**

Package.json .exports field is not correctly specified.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-7.html#packagejson-exports-imports-and-self-referencing notes that typescript will look relative to the import/require file for a .d.ts and ignore the types field when exports is used (I don't think these docs explicitly say this, but, the below error does). 

SVGO does not specify the type file in its `package.exports` field:

```
"main": "./lib/svgo-node.js",
  "bin": "./bin/svgo.js",
  "types": "./lib/svgo.d.ts",
  "exports": {
    "require": "./dist/svgo-node.cjs",
    "default": "./lib/svgo-node.js"
  },
```

I think this means TS is looking at either `./dist/svgo-node.d.ts` or `./lib/svgo-node.d.ts`, and the following error seems to hint that TS _can_ find the types but it really wants the package.json specified properly:

```
  λ  npx tsc --noEmit
src/filename.ts:5:26 - error TS7016: Could not find a declaration file for module 'svgo'. '/Users/.../node_modules/svgo/lib/svgo-node.js' implicitly has an 'any' type.
  There are types at '/Users/..../node_modules/svgo/lib/svgo.d.ts', but this result could not be resolved when respecting package.json "exports". The 'svgo' library may need to update its package.json or typings.

5 import { optimize } from 'svgo'
                           ~~~~~~

Found 1 error in src/filename.ts:5
```

I expect this is related to my own module specification and other typescript config:

```
{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Client",
  "compilerOptions": {
    "esModuleInterop": true,
    "skipLibCheck": true,
    "target": "es2022",
    "allowJs": true,
    "resolveJsonModule": true,
    "moduleDetection": "force",
    "isolatedModules": true,
    "verbatimModuleSyntax": false,

    "strict": true,
    "noUncheckedIndexedAccess": false,
    "noImplicitOverride": true,

    "composite": false,
    "declaration": true,
    "declarationMap": true,
    "sourceMap": true,

    "module": "preserve",
    "noEmit": true,

    "lib": ["es2022", "dom", "dom.iterable"],
    "jsx": "react-jsx"
  },
  "exclude": ["node_modules"]
}
```

Note that this the same config Matt Pocock uses and is documented [here](https://www.totaltypescript.com/tsconfig-cheat-sheet#building-for-a-library-in-a-monorepo). 

I'm sure many configurations of TS maybe will mask this bug, but, this setup seems pretty barebones and the linked definition for what exports should look like is fairly well defined.

- TS version: 5.4.5.
- Node version 20.12.2.
- SVGO version: 3.3.0..

**To Reproduce**

Run `tsc` across a file that imports from 'svgo'.
Its quite possible that different specifications for TS mask or otherwise do not exhibit the error.

**Expected behavior**

TS should be able to find the types.
It does mention that using 1 type file across ESM/CJS could potentially lead to problems.

**Desktop (please complete the following information):**

- TS version: 5.4.5.
- Node version 20.12.2.
- SVGO version: 3.3.0..

**Comment by mattstyles** [comment]
https://github.com/svg/svgo/pull/1984/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

#1984 will fix this issue.

**Comment by SethFalco** [comment]
Sorry for the headache! Indeed, we've merged that PR and made a release now, so it should be resolved in v3.3.1.

Thanks for reporting the issue and providing insight meanwhile.
